egs/tedlium/s5/local/chain/run_tdnn.sh

```diff
@@ -0,0 +1,205 @@
+fi
+
+if [ $stage -le 12 ]; then
+  echo "$0: creating neural net configs";
+
+  # create the config files for nnet initialization
+  repair_opts=${self_repair_scale:+" --self-repair-scale $self_repair_scale "}
+
+  steps/nnet3/tdnn/make_configs.py \
+    $repair_opts \
+    --feat-dir data/${train_set}_sp_hires \
+    --ivector-dir exp/nnet3/ivectors_${train_set}_sp \
+    --tree-dir $treedir \
+    --relu-dim $relu_dim \
+    --splice-indexes "-1,0,1 -1,0,1,2 -3,0,3 -3,0,3 -3,0,3 -6,-3,0 0" \
+    --use-presoftmax-prior-scale false \
+    --xent-regularize $xent_regularize \
+    --xent-separate-forward-affine true \
+    --include-log-softmax false \
+    --final-layer-normalize-target $final_layer_normalize_target \
+    $dir/configs || exit 1;
+fi
+
+if [ $stage -le 13 ]; then
+  if [[  $(hostname -f) ==  *.clsp.jhu.edu ]]; then
+     # spread the egs over various machines.  will help reduce overload of any
+     # one machine.
+     utils/create_split_dir.pl /export/b{09,10,11,12}/$USER/kaldi-data/egs/tedlium-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
+  fi
+
+  touch $dir/egs/.nodelete
+
+ steps/nnet3/chain/train.py --stage $train_stage \
+   --cmd "$decode_cmd" \
+   --feat.online-ivector-dir exp/nnet3/ivectors_${train_set}_sp \
+   --feat.cmvn-opts "--norm-means=false --norm-vars=false" \
+   --chain.xent-regularize $xent_regularize \
+   --chain.leaky-hmm-coefficient 0.1 \
+   --chain.l2-regularize 0.00005 \
+   --chain.apply-deriv-weights false \
+   --chain.lm-opts="--num-extra-lm-states=2000" \
+   --egs.stage $get_egs_stage \
+   --egs.opts "--frames-overlap-per-eg 0" \
+   --egs.chunk-width $frames_per_eg \
+   --trainer.num-chunk-per-minibatch $minibatch_size \
+   --trainer.frames-per-iter 1500000 \
+   --trainer.num-epochs $num_epochs \
+   --trainer.optimization.num-jobs-initial $num_jobs_initial \
+   --trainer.optimization.num-jobs-final $num_jobs_final \
+   --trainer.optimization.initial-effective-lrate $initial_effective_lrate \
+   --trainer.optimization.final-effective-lrate $final_effective_lrate \
+   --trainer.max-param-change $max_param_change \
+   --cleanup.remove-egs $remove_egs \
+   --cleanup.preserve-model-interval 20 \
+   --feat-dir data/${train_set}_sp_hires \
+   --tree-dir $treedir \
+   --lat-dir $lats_dir \
+   --dir $dir || exit 1;
+fi
+
+if [ $stage -le 14 ]; then
+  # Note: it might appear that this $lang directory is mismatched, and it is as
+  # far as the 'topo' is concerned, but this script doesn't read the 'topo' from
+  # the lang directory.
+  utils/mkgraph.sh --self-loop-scale 1.0 data/lang_test $dir $dir/graph
+fi
+
+graph_dir=$dir/graph
+if [ $stage -le 15 ]; then
+  iter_opts=
+  if [ ! -z $decode_iter ]; then
+    iter_opts=" --iter $decode_iter "
+  fi
+
+  for decode_set in dev test; do
+    (
+    steps/nnet3/decode.sh --acwt 1.0 --post-decode-acwt 10.0 \
+      --nj 8 --cmd "$decode_cmd" $iter_opts \
```

Comment:
It would be better to determine the number of jobs based on the number of speakers in the dataset.